concordium-contracts-common-derive/src/lib.rs

```diff
@@ -473,9 +488,27 @@ fn impl_serial(ast: &syn::DeriveInput) -> syn::Result<TokenStream> {
         _ => unimplemented!("#[derive(Serial)] is not implemented for union."),
     };
 
+    let (impl_generics, ty_generics, where_clauses) = ast.generics.split_for_impl();
+    // Extend where clauses with Serial predicate of each generic.
+    let where_clause_serial: proc_macro2::TokenStream = ast
+        .generics
+        .type_params()
+        .map(|type_param| {
+            let type_param_ident = &type_param.ident;
+            quote! (#type_param_ident: #root::Serial,)
+        })
+        .collect();
+
+    let where_clauses_tokens = if let Some(where_clauses) = where_clauses {
+        let predicates = &where_clauses.predicates;
+        quote!(#predicates, where_clause_serial)
+    } else {
+        where_clause_serial
+    };
```

Comment:
I think the implementation is fine as it is if we always want bounds for serial/deserial. 

But as we discussed I think it would be useful to be able to specify manual bounds like with `serde(bound)` (https://serde.rs/attr-bound.html )

Is there any problem with that?